notebooks/cugraph_benchmarks/sssp_benchmark.ipynb

```diff
@@ -0,0 +1,415 @@
+    "os.environ[\"CUDA_VISIBLE_DEVICES\"]=\"0\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Now load the required libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import needed libraries\n",
+    "import gc\n",
+    "import time\n",
+    "import rmm\n",
+    "import cugraph\n",
+    "import cudf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NetworkX libraries\n",
+    "import networkx as nx\n",
+    "from scipy.io import mmread"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try: \n",
+    "    import matplotlib\n",
+    "except ModuleNotFoundError:\n",
+    "    os.system('pip install matplotlib')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt; plt.rcdefaults()\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Test File\n",
+    "data = {\n",
+    "    'preferentialAttachment' : './data/preferentialAttachment.mtx',\n",
+    "    'caidaRouterLevel'       : './data/caidaRouterLevel.mtx',\n",
+    "    'coAuthorsDBLP'          : './data/coAuthorsDBLP.mtx',\n",
+    "    'dblp'                   : './data/dblp-2010.mtx',\n",
+    "    'citationCiteseer'       : './data/citationCiteseer.mtx',\n",
+    "    'coPapersDBLP'           : './data/coPapersDBLP.mtx',\n",
+    "    'coPapersCiteseer'       : './data/coPapersCiteseer.mtx',\n",
+    "    'as-Skitter'             : './data/as-Skitter.mtx'\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the testing functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Data reader - the file format is MTX, so we will use the reader from SciPy\n",
+    "def read_mtx_file(mm_file):\n",
+    "    print('Reading ' + str(mm_file) + '...')\n",
+    "    M = mmread(mm_file).asfptype()\n",
+    "     \n",
+    "    return M"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# CuGraph SSSP\n",
+    "\n",
+    "def cugraph_call(M):\n",
+    "\n",
+    "    gdf = cudf.DataFrame()\n",
+    "    gdf['src'] = M.row\n",
+    "    gdf['dst'] = M.col\n",
+    "    \n",
+    "    print('\\tcuGraph Solving... ')\n",
+    "    \n",
+    "    t1 = time.time()\n",
+    "        \n",
+    "    # cugraph SSSP Call\n",
+    "    G = cugraph.Graph()\n",
```

Comment:
Since we run sssp for nx DiGraph, we might want to compare time with cugraph DiGraph

Reply:
updated to DiGraph